Userland/Libraries/LibGfx/ImageFormats/WebPWriter.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2024, Nico Weber <thakis@chromium.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <LibGfx/Forward.h>
+
+namespace Gfx {
+
+struct WebPEncoderOptions {
+    Optional<ReadonlyBytes> icc_data;
+};
```

Comment:
I understand the scaffolding first and meat afterward approach. But I don't know if that's this helpful here, as you do not even implement it in this PR. (It's not a huge deal either, so no change requested)

Reply:
I implemented it in this PR at first, but I realized my code was broken and incomplete. The correct implementation is somewhat big (94 insertions(+), 3 deletions(-) -- most of that code is needed for animations too), so I figured I'd put it in a follow-up.